core/src/avm2/object/xml_list_object.rs

```diff
@@ -230,6 +328,79 @@ impl<'a, 'gc> Deref for E4XWrapper<'a, 'gc> {
     }
 }
 
+/// Represents either a XmlObject or a XmlListObject. Used
+/// for resolving the value of empty XMLLists.
+#[derive(Clone, Collect, Copy, Debug)]
+#[collect(no_drop)]
+pub enum XmlOrXmlListObject<'gc> {
```

Comment:
personally I'd probably keep using an Object, but I guess this is slightly safer.